pkg/aws/networkpool/validate_networkpool.go

```diff
@@ -0,0 +1,158 @@
+package networkpool
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/v2/pkg/apis/infrastructure/v1alpha2"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+
+	"github.com/giantswarm/aws-admission-controller/config"
+	"github.com/giantswarm/aws-admission-controller/pkg/validator"
+)
+
+type Validator struct {
+	dockerCIDR               string
+	ipamNetworkCIDR          string
+	k8sClient                k8sclient.Interface
+	kubernetesClusterIPRange string
+	logger                   micrologger.Logger
+}
+
+func NewValidator(config config.Config) (*Validator, error) {
+	if config.DockerCIDR == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.DockerCIDR must not be empty", config)
+	}
+	if config.IPAMNetworkCIDR == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.IPAMNetworkCIDR must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.KubernetesClusterIPRange == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.KubernetesClusterIPRange must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	validator := &Validator{
+		dockerCIDR:               config.DockerCIDR,
+		ipamNetworkCIDR:          config.IPAMNetworkCIDR,
+		k8sClient:                config.K8sClient,
+		kubernetesClusterIPRange: config.KubernetesClusterIPRange,
+		logger:                   config.Logger,
+	}
+
+	return validator, nil
+}
+
+func (v *Validator) Validate(request *v1beta1.AdmissionRequest) (bool, error) {
+	var networkPool infrastructurev1alpha2.NetworkPool
+	var allowed bool
+
+	if _, _, err := validator.Deserializer.Decode(request.Object.Raw, nil, &networkPool); err != nil {
+		return false, microerror.Maskf(parsingFailedError, "unable to parse networkpool: %v", err)
+	}
+	allowed, err := v.networkPoolOverlapping(networkPool)
```

Comment:
When I read `networkPoolOverlapping` I expect it to return `true` in case the networks are overlapping. Assigning `true` to `allowed` would be ok if the validation wants to ensure that network pools are overlapping, though my understanding so far was that we want to ensure that network pools do not overlap. Something is off here.

Reply:
I think the wording doesn't make sense, I can change name of the function to `networkPoolAllowed`, would this be better?